src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1542,15 +1555,20 @@ FixedwingPositionControl::control_auto_takeoff(const hrt_abstime &now, const flo
 			float target_airspeed = get_auto_airspeed_setpoint(now, _param_fw_airspd_trim.get(), ground_speed,
 						dt);
 
+			Vector2f curr_pos_local{_local_pos.x, _local_pos.y};
+			Vector2f curr_wp_local = _global_local_proj_ref.project(curr_wp(0), curr_wp(1));
+			Vector2f prev_wp_local = _global_local_proj_ref.project(_runway_takeoff.getStartWP()(0),
```

Comment:
Launch-based take-off doesn't use runway takeoff starting waypoint, uses `prev_wp`

Reply:
Ah thanks for pointing this out!